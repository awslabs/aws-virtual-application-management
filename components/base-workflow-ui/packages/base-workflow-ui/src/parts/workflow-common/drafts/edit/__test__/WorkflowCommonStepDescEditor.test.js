/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 *
 * Licensed under the Apache License, Version 2.0 (the "License").
 * You may not use this file except in compliance with the License.
 * A copy of the License is located at
 *
 * http://aws.amazon.com/apache2.0
 *
 * or in the "license" file accompanying this file. This file is distributed
 * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
 * express or implied. See the License for the specific language governing
 * permissions and limitations under the License.
 */

import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import { Provider } from 'mobx-react';
import WorkflowCommonStepDescEditor from '../WorkflowCommonStepDescEditor';

describe('WorkflowCommonStepDescEditor', () => {
  // eslint-disable-next-line no-unused-vars
  let component = null;
  let wrapper = null;
  const onCancel = jest.fn();
  const editor = { version: { selectedSteps: [] }, addStep: jest.fn(), stepEditorsEditing: jest.fn() };
  const stepEditor = { step: { desc: 'test-desc' } };
  const stepTemplatesStore = { load: jest.fn() };
  beforeEach(() => {
    // Render component
    wrapper = shallow(
      <Provider stepTemplatesStore={stepTemplatesStore}>
        <WorkflowCommonStepDescEditor
          editor={editor}
          stepEditor={stepEditor}
          onCancel={onCancel}
          StepTemplatesStore={stepTemplatesStore}
        />
      </Provider>,
    );
    // Get instance of the component
    component = wrapper.instance();
  });
  it('renders correctly', () =>
    expect(
      renderer
        .create(
          <Provider stepTemplatesStore={stepTemplatesStore}>
            <WorkflowCommonStepDescEditor
              editor={editor}
              stepEditor={stepEditor}
              onCancel={onCancel}
              StepTemplatesStore={stepTemplatesStore}
            />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot());
});
