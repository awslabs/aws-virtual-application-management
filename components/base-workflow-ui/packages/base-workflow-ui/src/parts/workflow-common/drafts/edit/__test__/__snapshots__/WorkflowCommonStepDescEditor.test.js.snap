// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WorkflowCommonStepDescEditor renders correctly 1`] = `
<div>
  <div
    className="flex animated"
  >
    <div
      className="flex-auto"
    >
      <i
        aria-hidden="true"
        className="file alternate outline icon mr1 color-grey"
        onClick={[Function]}
      />
      <b>
        Description
      </b>
    </div>
    <div
      className="pl1 pr0"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="grey edit icon cursor-pointer"
        onClick={[Function]}
      />
    </div>
  </div>
  <div
    className="ui divider mt1 mb2"
  />
  <div
    className="animated"
  >
    <p>
      test-desc
    </p>
  </div>
</div>
`;
